### ISSUE TITLE
`doc.modifiedPaths()` returns duplicate values for sub schema docs

### ISSUE THREAD

**Issue by adambuczynski** [Automattic/mongoose#4223]
I have a sub schema in a document, and if I query the document for `modifiedPaths()`, I get duplicate data for every path to one of the sub document properties. Is this intentional or a bug?

Example output:
```js
[ 'description',
  'organisers',
  'courts',
  'recurrence',
  'recurrence.amount',
  'recurrence',
  'recurrence.lastDate',
  'recurrence',
  'recurrence.firstDate',
  'recurrence',
  'recurrence.monthlyDayOf',
  'recurrence',
  'recurrence.ends',
  'recurrence',
  'recurrence.weekdays',
  'recurrence',
  'recurrence.frequency',
  'recurrence',
  'recurrence.interval' ]
```

Should the return array be filtered for unique values? If yes, I can make a PR for it.

**Comment by vkarpov15** [comment]
Looks like a bug, sure, PRs welcome

**Comment by adambuczynski** [comment]
Closed via #4226
